scripts/initialize-build-host.sh

```diff
@@ -153,11 +178,34 @@ then
     fi
 
     # --------------------------------------------------------------------------
-    # Populate build host.
+    # Check target machine health.
     # --------------------------------------------------------------------------
 
     login="$(cat $HOME/proxy-target.txt)"
 
+    if [ -z "$login" ]
+    then
+        echo "Proxy target address not available, aborting the build"
+        exit 1
+    fi
+
+    if $RSH $login true
+    then
+	:
+    else
+	if [ -f $HOME/on-vm-hypervisor ]
+            echo "Failed to SSH to nested VM, probably it's hanging, resetting it"
+            reset_nested_vm
+        else
+            echo "Failed to SSH to proxy target, aborting the build"
+        fi
```

Comment:
Aborting, so do you want to call exit here?

Reply:
indeed!